### PR TITLE
Allow additional path parameters to be added for capabilities validation

### PR DESCRIPTION
**Pull request by thecoop**
Due to https://github.com/elastic/elasticsearch/issues/36785, path params are also present in `params` set. So we need to add any path params to the supported set we check against, but that are not published by the capabilities API.

Define an extra method to do this, so marking those that need to be refactored when we re-visit params validation.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/rest/RestHandler.java [hunk 1]
```diff
@@ -86,6 +86,19 @@ default List<Route> routes() {
...
 
+    /**
+     * The set of path and query parameters that could be present on this handler.
+     * This method is only required due to https://github.com/elastic/elasticsearch/issues/36785,
+     * which conflates query and path parameters inside the rest handler.
+     * This method should be overridden to add path parameters to {@link #supportedQueryParameters}
+     * if the handler has path parameters.
+     * This method will be removed when {@link #supportedQueryParameters()} and {@link BaseRestHandler#responseParams()} are merged.
+     */
+    @Deprecated(forRemoval = true)
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/rest/RestHandler.java:97]: Hmm this puts a red squiggly under overriding implementations of this method as if you're doing something wrong by touching it, but as things stand today it's required. Could we not deprecate it like this until we're ready?

**Review by DaveCTurner — APPROVED** [review]
LGTM except for the deprecation annotation.

**Comment by thecoop** [comment]
@elasticmachine update branch

**Review by mosche — APPROVED** [review]
lgtm

**Pull request merged**